features/flows/kafka-persistence/src/main/java/org/opennms/netmgt/flows/persistence/KafkaFlowForwarder.java

```diff
@@ -0,0 +1,139 @@
+import com.swrve.ratelimitedlogger.RateLimitedLog;
+
+public class KafkaFlowForwarder implements EnrichedFlowForwarder {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaFlowForwarder.class);
+    public static final String KAFKA_CLIENT_PID = "org.opennms.features.flows.persistence.kafka";
+    private final ConfigurationAdmin configAdmin;
+    private KafkaProducer<String, byte[]> producer;
+    private String topicName;
+    private static final RateLimitedLog RATE_LIMITED_LOG = RateLimitedLog
+            .withRateLimit(LOG)
+            .maxRate(1).every(Duration.standardSeconds(60))
+            .build();
+    private Properties producerConfig;
+    
+    public KafkaFlowForwarder(ConfigurationAdmin configAdmin) {
+        this.configAdmin = configAdmin;
+    }
+
+    @Override
+    public void forward(EnrichedFlow enrichedFlow) {
+
+        if (producer == null) {
+            RATE_LIMITED_LOG.warn("Kafka Producer is not configured for flow forwarding.");
+            return;
+        }
+
+        try {
+            FlowDocument flowDocument = FlowDocumentBuilder.buildFlowDocument(enrichedFlow);
+            final ProducerRecord<String, byte[]> record = new ProducerRecord<>(topicName, flowDocument.toByteArray());
+            producer.send(record, (recordMetadata, e) -> {
+                if (e != null) {
+                    RATE_LIMITED_LOG.warn("Failed to send flow document to producer: {}.", record, e);
+                } else if (LOG.isTraceEnabled()) {
+                    LOG.trace("Persisted flow document {} to kafka.", flowDocument);
+                }
+            });
+        } catch (Exception e) {
+            LOG.error("Exception while sending flow to kafka", e);
+        }
+    }
+
+    public void init() throws IOException {
+        // Create the Kafka producer
+        producerConfig = new Properties();
+        final Dictionary<String, Object> properties = configAdmin
+                .getConfiguration(KAFKA_CLIENT_PID).getProperties();
+        if (properties != null) {
+            final Enumeration<String> keys = properties.keys();
+            while (keys.hasMoreElements()) {
+                final String key = keys.nextElement();
+                producerConfig.put(key, properties.get(key));
+            }
+        }
+
+        if (producerConfig.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG) != null) {
+            // Overwrite the serializers
+            producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getCanonicalName());
+            producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getCanonicalName());
+            // Class-loader hack for accessing the kafka classes when initializing producer.
+            producer = runWithGivenClassLoader(() -> new KafkaProducer<>(producerConfig), KafkaProducer.class.getClassLoader());
+            LOG.info(" Kafka Producer initialized with config {} ", producerConfig);
```

Comment:
```suggestion
            LOG.info("Kafka Producer initialized with config: {}", producerConfig);
```

Remove leading and trailing spaces from log message.